test/integration/good/compiler-optimizations/dune

```diff
@@ -74,6 +75,8 @@
    fails-test.stan
    off-small.stan
    off-dce.stan
+   overloaded-fn.stan
+   overloaded-fn2.stan
```

Comment:
Apparently I don't how these tests work. Why isn't this list just `(glob_files *.stan)`? And was missing `overloaded-fn.stan` a bug in the test?

Reply:
I think it was a historical thing, where some of the optimizations could cause the compiler to crash so we didn’t run all of them? Not completely sure, it predates me on the project. 

I wouldn’t be opposed to just making them a glob like the other tests are

Reply:
Looked through the blame, and it seems that at first there was only one file in this directory and the dune file was written without the glob, and that seems to have just persisted without much interrogation.